Pathogens/Utility/location_utils/mapbox_api_client.py

```diff
@@ -0,0 +1,93 @@
+import os
+import requests
+
+from dataclasses import dataclass
+from Pathogens.Utility.location_utils.city_state_country_mapbox_response_data_cache import update_mapbox_response_cache, attempt_to_fetch_mapbox_response_data_from_cache
+from Pathogens.Utility.location_utils.country_codes import get_country_code, read_from_json
+
+MAPBOX_REQUEST_PARAM_OVERRIDES = read_from_json('mapbox_request_param_overrides.json')
+
+@dataclass
+class MapboxResponse:
+    center_coordinates: [float, float]
+    bounding_box: [float, float, float, float]
+    text: str
+
+@dataclass
+class MapboxRequestParams:
+    mapbox_search_text: str
+    country_code: str
+    geocoder_data_type: str
+
+def get_mapbox_api_query_url(mapbox_request_params: MapboxRequestParams):
+    return f"https://api.mapbox.com/geocoding/v5/mapbox.places/{mapbox_request_params.mapbox_search_text}.json?" \
+          f"access_token={os.getenv('MAPBOX_API_KEY')}&types={mapbox_request_params.geocoder_data_type}&country={mapbox_request_params.country_code}"
+
+def parse_mapbox_response(response):
+    data = response.json()
+    if data and "features" in data and len(data['features']) > 0:
+        return MapboxResponse(
+            center_coordinates = data['features'][0]['center'],
+            bounding_box = data['features'][0].get('bbox', None),
+            text = data['features'][0].get('text', None)
+        )
+    else:
+        return None
+
+def generate_mapbox_request_params(city_name: str | None, state_name: str | None, country_name: str):
+    country_code = get_country_code(country_name=country_name, iso3=False)
+
+    mapbox_request_param_override = MAPBOX_REQUEST_PARAM_OVERRIDES.get(country_code, {}) \
+        .get(state_name.strip() if state_name is not None else 'N/A', {}) \
+        .get(city_name.strip() if city_name is not None else 'N/A', None)
+
+    if(mapbox_request_param_override is not None):
+        return MapboxRequestParams(
+            mapbox_search_text = mapbox_request_param_override['mapbox_search_text'],
+            country_code = mapbox_request_param_override['country_code'],
+            geocoder_data_type = mapbox_request_param_override['geocoder_data_type']
+        )
```

Comment:
Meat and potatoes alert ⚠️

Basically takes the wheel away from my code for specific data points which I know are in a format that mapbox won't like.